utils/s2n_safety.h

```diff
@@ -53,6 +53,11 @@
  */
 #define ENSURE( condition , error )                  __S2N_ENSURE((condition), BAIL(error))
 
+/**
+ * Ensures the `condition` is `true`, otherwise the function will `BAIL` with an `error`
+ */
+#define ENSURE_CONDITION( condition , error )        __S2N_ENSURE_CONDITION((condition), BAIL(error))
```

Comment:
I don't think we want to have `ENSURE_CONDITION` called directly, as It's not super clear why it's different from `ENSURE`.

I would say remove `ENSURE_CONDITION` and `ENSURE_POSIX_CONDITION` and define `PRECONDITION` and `POSTCONDITION` in terms of `__S2N_ENSURE_CONDITION`.

Reply:
Now, both `PRECONDITION` and `POSTCONDITION` are in term of `__S2N_ENSURE_CONDITION`.